src/dfx/src/lib/sns/deploy.rs

```diff
@@ -1,30 +1,54 @@
 //! Code for creating an SNS.
-use anyhow::bail;
+use anyhow::{anyhow, bail, Context};
 use fn_error_context::context;
 use std::ffi::OsString;
 use std::path::Path;
 
 use crate::lib::call_bundled::call_bundled;
 use crate::lib::error::DfxResult;
+use crate::lib::models::canister_id_store::CanisterIdStore;
+use crate::lib::provider::create_agent_environment;
 use crate::Environment;
 
 /// Creates an SNS.  This requires funds but no proposal.
 #[context("Failed to deploy SNS with config: {}", path.display())]
-pub fn deploy_sns(env: &dyn Environment, path: &Path) -> DfxResult<String> {
+pub fn deploy_sns(
+    env: &dyn Environment,
+    path: &Path,
+    network: Option<String>,
+) -> DfxResult<String> {
+    // For networks other than ic and local we need a provider URL:
+    let agent_environment = create_agent_environment(env, network.clone())?;
+    let network_descriptor = agent_environment.get_network_descriptor();
```

Comment:
Since we're only using the agent environment to get the network descriptor, we can instead create the network descriptor directly.

```
        let network_descriptor = create_network_descriptor(
            env.get_config(),
            env.get_networks_config(),
            network,
            None,
            LocalBindDetermination::ApplyRunningWebserverPort,
        )?;
```